packages/storage/src/implementation/requests.ts

```diff
@@ -104,7 +105,13 @@ export function sharedErrorHandler(
   ): FirebaseStorageError {
     let newErr;
     if (xhr.getStatus() === 401) {
-      newErr = unauthenticated();
+      if (
+        xhr.getResponseText().includes('Firebase App Check token is invalid')
```

Comment:
Is there a code that we can check against?

Reply:
We could make this string into a local constant but it would have to be manually kept in sync. This is where the string comes from in the backend, we don't have any access to it here. The status code is 401 which is shared with some other errors. https://source.corp.google.com/piper///depot/google3/java/com/google/firebase/storage/authorization/appcheck/AppCheckAuthorizer.java;rcl=369703237;l=26

Reply:
I was hoping server sends back a json object with a code property that we can check against. The error text is subject to change in nature, so I'm worried we will get out of sync at one point.

Reply:
It doesn't seem like anything consistent is sent back except the error code and that string https://source.corp.google.com/piper///depot/google3/java/com/google/firebase/storage/authorization/appcheck/AppCheckAuthorizer.java;rcl=370565192;l=78

Reply:
Hm. Can you add a comment to the above linked file to state that the text of the error message cannot be changed?

Reply:
Added a comment.